frontend-dsl/spigot/src/main/groovy/SpigotModsDotGroovy.groovy

```diff
@@ -0,0 +1,189 @@
+     * <p>This <i>must</i> contain the full namespace including the class file itself.</p>
+     */
+    String main
+
+    /**@
+     * The name of your plugin.
+     * <p><i>Must</i> consist of all alphanumeric characters and underscores {@code (a-z,A-Z,0-9, _)}</p>
+     */
+    String name
+
+    /**@
+     * The version of your plugin.
+     * <p>The most common versioning format is <a href="https://semver.org/">Semantic Versioning</a>, which is written
+     * as MAJOR.MINOR.PATCH (eg: 1.4.1 or 9.12.4)</p>
+     */
+    String version
+
+    /**@
+     * A human friendly description of the functionality your plugin provides.
+     * <p>The description can have multiple lines.</p>
+     */
+    @Nullable String description = null
+
+    /**@
+     * The version of the API you want to use.
+     * <p>1.13, 1.14, 1.15, 1.16, 1.17, 1.18, 1.19 and 1.20 are the available versions.</p>
+     * <p>This will signal to the server that your plugin has been coded with a specific server version in mind, and
+     * that it should not apply any sort of backwards compatibility measures. As a result you will also need to make
+     * sure that you have programmed your code to account for reading of older configurations, data, etc...</p>
+     * <p>Each server version can decide how compatibility is achieved, unknown or future versions will prevent the
+     * plugin from enabling. As of the 1.14 release, the api-version 1.13 is still allowed - however future versions
+     * may drop backwards support based on this version.</p>
+     */
+    @Nullable String apiVersion = null
+
+    /**@
+     * Explicitly state when a plugin should be loaded. If not supplied will default to {@link LoadState#POSTWORLD}.
+     */
+    @Nullable LoadState load = LoadState.POST_WORLD
+
+    /**@
+     * Uniquely identifies who developed this plugin.
+     * <p>Used in some server error messages to provide helpful feedback on who to contact when an error occurs.</p>
+     * <p>A <a href="https://spigotmc.org">SpigotMC.org</a> forum handle or email address is recommended.</p>
+     */
+    @Nullable String author = null
+
+    /**@
+     * Allows you to list multiple authors, if it is a collaborative project.
+     */
+    @Nullable List<String> authors = null
+
+    /**@
+     * The plugin's or author's website.
+     * <p>If you have no dedicated website, a link to the page where this plugin is listed is recommended.</p>
+     */
+    @Nullable String website = null
+
+    /**@
+     * The name to use when logging to console instead of the plugin's name.
+     */
+    @Nullable String prefix = null
+
+    /**@
+     * A list of plugin names that your plugin requires to load.
+     * <p>If any plugin listed here is not found your plugin will fail to load.</p>
+     * <p>If multiple plugins list each other as a depend, so that there are no plugins without an unloadable
+     * dependency, all will fail to load.</p>
+     */
+    @Nullable List<String> depend = null
```

Comment:
I would name it `dependencies`, `depend` doesn't sit right with me. Maybe we could introduce an aliasing system (annotation-based) so that the name is correct in the map?

Reply:
The naming of properties for each platform's frontend is deliberately the same as [its official spec](https://www.spigotmc.org/wiki/plugin-yml/#:~:text=www.spigotmc.org-,depend,-A%20list%20of), however having an alias system is a good idea! In the meantime I'll consider adding a `setDependencies(List<String>)` method for an alias